### PR TITLE
Feature/75240 ui remove a tag from scope

### PR DESCRIPTION
**Pull request by arildbjerkenes**
Ideally, backend should have a bulk endpoint to remove tags. We don't have this so we just delete tag by tag.

### CODE REVIEW

**Comment by arildbjerkenes** [comment]
NOTE: I made a mistake, by not starting from master branch... I think it will work fine.

File: src/modules/Preservation/views/EditTagProperties/EditTagProperties.tsx [hunk 1]
```diff
@@ -344,6 +355,9 @@ const EditTagProperties = (): JSX.Element => {
                 :
                 <Container>
                     <div>
+                        <ErrorContainer>
+                            {validationErrorMessage && (<Typography variant="caption">{validationErrorMessage}</Typography>)}
+                        </ErrorContainer>
```
Comment (Cnordbo) [inline src/modules/Preservation/views/EditTagProperties/EditTagProperties.tsx:360]: Add `ErrorContainer` to the conditional render.

File: src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx [hunk 2]
```diff
@@ -382,6 +382,50 @@ const ScopeOverview: React.FC = (): JSX.Element => {
...
+                rowVersion: t.rowVersion
+            })));
+            refreshScopeList();
+            setSelectedTags([]);
+            showSnackbarNotification('Selected tag(s) have been removed.');
+        } catch (error) {
+            console.error('Remove failed: ', error.message, error.data);
+            showSnackbarNotification(error.message);
+        }
+        return Promise.resolve();
```
Comment (Cnordbo) [inline src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx:401]: Not needed.

File: src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx [hunk 3]
```diff
@@ -382,6 +382,50 @@ const ScopeOverview: React.FC = (): JSX.Element => {
             completeFunc);
     };
 
+    let removableTags: PreservedTag[] = [];
+    let nonRemovableTags: PreservedTag[] = [];
```
Comment (Cnordbo) [inline src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx:386]: Make these local to `showRemoveDialog` and send them in to `remove` as parameters from the remove dialog.

File: src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx [hunk 4]
```diff
@@ -382,6 +382,50 @@ const ScopeOverview: React.FC = (): JSX.Element => {
...
+        selectedTags.map((tag) => {
+            const newTag: PreservedTag = { ...tag };
+            if (tag.isVoided && !tag.isInUse) {
+                removableTags.push(newTag);
+            } else {
+                nonRemovableTags.push(newTag);
+            }
+        });
+        const removeButton = removableTags.length > 0 ? 'Remove' : null;
+        const removeFunc = removableTags.length > 0 ? remove : null;
```
Comment (Cnordbo) [inline src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx:417]: Solution when changing the "global" removableTags/nonremovabletags to local variables. 

```js
        const removeFunc = removableTags.length > 0 ? () => remove(removableTags) : null;
```

**Review by Cnordbo — CHANGES_REQUESTED** [review]

File: src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx (hunk 3 above)
Comment (arildbjerkenes) [reply src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx:386]: I know this is not a good solution, but all the other dialogs are implemented like this. The whole impl could need a refactoring. But we don't have time for this now. I think for this to work we would need to refactor showModelDialog, right? The callback functions do not have parameters.

Comment (Cnordbo) [reply src/modules/Preservation/views/ScopeOverview/ScopeOverview.tsx:386]: No, no large refactoring required, solution is proposed here: 

https://github.com/equinor/procosys-js-frontend/pull/243#discussion_r480030609

**Review by Cnordbo — APPROVED** [review]

**Pull request merged**